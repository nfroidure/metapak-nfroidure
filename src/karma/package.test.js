'use strict';

const packageTransformer = require('./package');

describe('Karma', () => {
  describe('Package transformer', () => {
    it('should work with some files', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['karma'],
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with mocha configs', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['mocha', 'karma'],
          },
        })
      ).toMatchSnapshot();
    });
  });
});
