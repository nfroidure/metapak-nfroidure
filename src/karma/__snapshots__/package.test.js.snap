// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Karma Package transformer should work with mocha configs 1`] = `
Object {
  "devDependencies": Object {
    "karma": "^3.0.0",
    "karma-chrome-launcher": "^2.2.0",
    "karma-firefox-launcher": "^1.1.0",
    "karma-mocha": "^1.3.0",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "karma",
      "karma-chrome-launcher",
      "karma-firefox-launcher",
      "karma-mocha",
    ],
  },
  "metapak": Object {
    "configs": Array [
      "mocha",
      "karma",
    ],
    "data": Object {
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "scripts": Object {
    "karma": "karma start karma.conf.js",
    "test": " && npm run karma",
  },
}
`;

exports[`Karma Package transformer should work with some files 1`] = `
Object {
  "devDependencies": Object {
    "karma": "^3.0.0",
    "karma-chrome-launcher": "^2.2.0",
    "karma-firefox-launcher": "^1.1.0",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "karma",
      "karma-chrome-launcher",
      "karma-firefox-launcher",
    ],
  },
  "metapak": Object {
    "configs": Array [
      "karma",
    ],
    "data": Object {
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "scripts": Object {
    "karma": "karma start karma.conf.js",
    "test": " && npm run karma",
  },
}
`;
